pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -268,52 +339,155 @@ func (p *DNSProxy) RemoveRestoredRules(endpointID uint16) {
 	p.removeRestoredRulesLocked(uint64(endpointID))
 }
 
+// lookupOrCompilePatternMatcher will check if the regex is already compiled and present in another policy, and
```

Comment:
Nit: 
```suggestion
// lookupOrCompilePatternMatcher will check if the pattern is already compiled and present in another policy, and
```

I think we should try to be consistent with our terms so that we don't cause unnecessary confusion to readers. 

* pattern == not yet compiled, string
* regex == compiled, object

Reply:
Could you please go through the rest of the changes and ensure that the proper term is used?

Reply:
Yeah, agree, thanks!. Will try to do that as best as I can 😅 Still not convinced what is easiest or best here, since `pattern` in the context of the `matchPattern` provided in a policy is different than the `pattern` for a regular expression. 

Internally in the go `regexp` package they call it both `pattern` and `expr` interchangeably (https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/regexp/regexp.go;l=85;bpv=1;bpt=1), stating `Regexp is the representation of a compiled regular expression.`.

Will try to make it consistent and use pattern for the non-compiled version, and regex for the compiled one 👍